app/front-end/src/features/editor/components/editorView/editorColumnMenu.tsx

```diff
@@ -1,39 +1,53 @@
 
 const StyledGridColumnMenuContainer = styled(GridColumnMenuContainer)(({ theme }) => ({
   backgroundColor: theme.palette.secondary.main,
 }));
 
-interface GridColumnMenuContainerProps extends GridColumnMenuProps {}
+interface GridColumnMenuContainerProps extends GridColumnMenuProps {
+  handleAggregation: (column: string, action: FileContentAggregationActions) => void;
+}
 
 /**
- * `EditorColumnMenu` component customizes the column menu in a DataGrid with a styled container.
+ * `EditorColumnMenu` component provides a custom column menu for data grid columns,
+ * allowing users to apply aggregations and hide columns.
  *
- * @description This component extends the default column menu functionality of the DataGrid by applying custom styles
- * to the menu container. The `StyledGridColumnMenuContainer` applies a background color from the theme's secondary palette
- * to the menu. The menu includes a `GridColumnMenuHideItem` for hiding the column, which invokes the `hideMenu` function
- * when clicked.
+ * @description
+ * The `EditorColumnMenu` is used within a `DataGrid` to offer additional options for column management.
+ * It includes options for applying different aggregation actions (e.g., sum, average) to the column data,
+ * and provides an option to hide the column.
  *
- * @component
- *
- * @param {GridColumnMenuContainerProps} props - The props for the component.
- * @param {() => void} props.hideMenu - A callback function to hide the column menu.
- * @param {object} props.colDef - Column definition object passed to the menu.
- * @param {GridColumnMenuProps} [other] - Other props that are passed to the `GridColumnMenuContainer`.
+ * - **Aggregation:** Users can select from various aggregation actions (sum, average, minimum, maximum, count) for the column.
+ * - **Hide Column:** Provides an option to hide the column from the data grid.
  *
- * @example
- * // Example usage of the EditorColumnMenu component
- * <EditorColumnMenu
- *   hideMenu={() => console.log('Hide menu')}
- *   colDef={columnDefinition}
- * />
+ * This component leverages `EditorColumnMenuAggregationItem` to render the aggregation options and `GridColumnMenuHideItem` for hiding the column.
  *
- * @returns {JSX.Element} A styled `GridColumnMenuContainer` containing a `GridColumnMenuHideItem`.
+ * @component
  */
-export const EditorColumnMenu: React.FC<GridColumnMenuContainerProps> = ({ hideMenu, colDef, ...other }) => {
+export const EditorColumnMenu: React.FC<GridColumnMenuContainerProps> = ({
+  handleAggregation,
+  hideMenu,
+  colDef,
+  ...other
+}) => {
+  const { fileContent } = useWorkspaceContext();
+  const aggregationActiveAction = fileContent.aggregations[colDef.field]
+    ? fileContent.aggregations[colDef.field].action
+    : FileContentAggregationActions.NONE;
+
   return (
     <StyledGridColumnMenuContainer hideMenu={hideMenu} colDef={colDef} {...other}>
```

Comment:
Container should be shrunk down in size since it's a lot larger than other elements.

Reply:
new bugfix PKFE-47